src/Aspire.Hosting.Azure.Kusto/api/Aspire.Hosting.Azure.Kusto.cs

```diff
@@ -0,0 +1,52 @@
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+namespace Aspire.Hosting.Azure.Kusto
+{
+    public static partial class AzureKustoBuilderExtensions
+    {
+        public static ApplicationModel.IResourceBuilder<AzureKustoClusterResource> AddAzureKustoCluster(this IDistributedApplicationBuilder builder, string name) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<AzureKustoDatabaseResource> AddDatabase(this ApplicationModel.IResourceBuilder<AzureKustoClusterResource> builder, string name, string? databaseName = null) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<AzureKustoClusterResource> RunAsEmulator(this ApplicationModel.IResourceBuilder<AzureKustoClusterResource> builder, System.Action<ApplicationModel.IResourceBuilder<AzureKustoEmulatorResource>>? configureContainer = null) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<AzureKustoDatabaseResource> WithCreationScript(this ApplicationModel.IResourceBuilder<AzureKustoDatabaseResource> builder, string script) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<AzureKustoEmulatorResource> WithHttpPort(this ApplicationModel.IResourceBuilder<AzureKustoEmulatorResource> builder, int port) { throw null; }
```

Comment:
Why isn't this something that would be configured as part of the callback to `RunAsEmulator`?

Reply:
#11520